images/agent/src/pkg/cache/cache.go

```diff
@@ -92,7 +93,7 @@ func (c *Cache) StoreLVs(lvs []internal.LVData, stdErr bytes.Buffer) {
 	}
 
 	for key, lv := range c.lvs {
-		if lv.Exist {
+		if lv.Exist && reflect.ValueOf(lv.Data).IsZero() {
```

Comment:
This loop is a cache cleanup procedure, and this condition was keeping non-removed records from being cleaned up. After this change, only zero-valued ones will be kept, which doesn't make sense.

Apart from this, couple of points regarding problems, that were there before:
- Line 100 - condition is excessive and can be removed, since all existent (`Exist: true`) items will be filtered on line 96, and non-existent should be deleted regardless of their presense in `lvsOnNode`
  - this refactor will let us remove `lvsOnNode` variable at all, we'll just use `lvs`
- Line 86 is excessive, because the key is already present as a for loop variable

Reply:
More complex existing problem, which came to my mind.

Since we're cleaning up removed records, we're losing information about them being removed. This information is helpful to avoid overwriting them with "somehow-late" store operations, see condition on the line 87, where we avoid overwriting `Exist: false` records.

So we'll end up with invalid cache records after some "late" operations. Solution would be to keep the timestamp of the entry removal and do a cleanup only for really old records (with no chance of catching "somehow late update operation")